src/sentry/integrations/opsgenie/integration.py

```diff
@@ -120,7 +120,46 @@ def dispatch(self, request: Request, pipeline) -> HttpResponse:  # type:ignore
 
 
 class OpsgenieIntegration(IntegrationInstallation):
-    pass
+    def get_client(self) -> Any:
+        return OpsgenieClient(integration=self.model)
+
+    def get_organization_config(self) -> Sequence[Any]:
+        fields = [
+            {
+                "name": "team_table",
+                "type": "table",
+                "label": "Opsgenie teams with the Sentry integration enabled",
+                "help": "If teams need to be updated, deleted, or added manually please do so here. Alert rules will need to be individually updated for any additions or deletions of teams.",
+                "addButtonText": "",
+                "columnLabels": {"team": "Team", "integration_key": "Integration Key"},
+                "columnKeys": ["team", "integration_key"],
+                "confirmDeleteMessage": "Any alert rules associated with this team will stop working. The rules will still exist but will show a `removed` team.",
+            }
+        ]
+
+        return fields
+
+    def update_organization_config(self, data: MutableMapping[str, Any]) -> None:
+        client = self.get_client()
+        # get the team ID/test the API key for a newly added row
+        teams = data["team_table"]
+        for team in teams:
+            if team["id"] != "":
+                continue
+            try:
+                resp = client.get_team_id(
+                    integration_key=team["integration_key"], team_name=team["team"]
+                )
+                team["id"] = resp["data"]["id"]
+            except ApiError:
+                raise ValidationError(
+                    {"api_key": ["Could not save: invalid team name or integration key."]}
```

Comment:
It may help throwing in the status code from Opsgenie. I imagine it's accessible from the raised `ApiError`.

Reply:
I feel like knowing the error code won't be too helpful for the user (I don't think most login failure screens, which I'm taking inspiration from, provide this), but let me know your reasoning!

Reply:
My thought process was only to indicate that this was the response we received from the integration, if they see 401, it could indicate they have outdated creds or something like that. It might be hard to convey that it isn't an error from our servers though.

That said, this is your project and If you don't prefer that user experience, it's yours to 🪓  it, i will leave it to you!